	- do not call uname -m, use passed ${ADK_ARCH} variable
	- use ${CC} instead of hard-coded gcc
--- libhugetlbfs-2.11.orig/Makefile	2010-12-16 18:38:22.000000000 +0100
+++ libhugetlbfs-2.11/Makefile	2011-01-22 21:09:48.008659615 +0100
@@ -31,51 +31,51 @@ CFLAGS ?= -O2 -g
 CFLAGS += -Wall -fPIC
 CPPFLAGS += -D__LIBHUGETLBFS__
 
-ARCH = $(shell uname -m | sed -e s/i.86/i386/)
+ARCH = $(shell echo ${ADK_ARCH} | sed -e s/i.86/i386/)
 
 ifeq ($(ARCH),ppc64)
-CC64 = gcc -m64
+CC64 = ${CC} -m64
 ELF64 = elf64ppc
 TMPLIB64 = lib64
 TMPLIB32 = lib
 ifneq ($(BUILDTYPE),NATIVEONLY)
-CC32 = gcc -m32
+CC32 = ${CC} -m32
 ELF32 = elf32ppclinux
 endif
 else
 ifeq ($(ARCH),ppc)
-CC32 = gcc -m32
+CC32 = ${CC} -m32
 ELF32 = elf32ppclinux
 TMPLIB32 = lib
 else
 ifeq ($(ARCH),i386)
-CC32 = gcc
+CC32 = ${CC}
 ELF32 = elf_i386
 TMPLIB32 = lib
 else
 ifeq ($(ARCH),x86_64)
-CC64 = gcc -m64
+CC64 = ${CC} -m64
 ELF64 = elf_x86_64
 TMPLIB64 = lib64
 TMPLIB32 = lib
 ifneq ($(BUILDTYPE),NATIVEONLY)
-CC32 = gcc -m32
+CC32 = ${CC} -m32
 ELF32 = elf_i386
 endif
 else
 ifeq ($(ARCH),ia64)
-CC64 = gcc
+CC64 = ${CC}
 TMPLIB64 = lib64
 CFLAGS += -DNO_ELFLINK
 else
 ifeq ($(ARCH),sparc64)
-CC64 = gcc -m64
+CC64 = ${CC} -m64
 TMPLIB64 = lib64
 CFLAGS += -DNO_ELFLINK
 else
 ifeq ($(ARCH),s390x)
-CC64 = gcc -m64
-CC32 = gcc -m31
+CC64 = ${CC} -m64
+CC32 = ${CC} -m31
 TMPLIB64 = lib64
 TMPLIB32 = lib
 CFLAGS += -DNO_ELFLINK
